BackendApartmentReservation/BackendApartmentReservation/Utilities/RegistrationInputValidator.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using BackendApartmentReservation.DataContracts.DataTransferObjects.Requests;
+using Castle.Core.Internal;
+
+namespace BackendApartmentReservation.Utilities
+{
+    public static class RegistrationInputValidator
+    {
+        public static bool IsInputValid(RegisterRequest registerRequest)
+        {
+            var isValidEmail = Regex.IsMatch(registerRequest.Email, @"\A(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?)\Z", RegexOptions.IgnoreCase);
```

Comment:
It's better not to reinvent the wheel with email regex :D take a look at my comment above